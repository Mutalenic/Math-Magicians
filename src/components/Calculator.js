import React from 'react';

export default class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="layout">
        <div className="print">0</div>
        <div className="keyboard">
          <button type="button" className="designcolor" id="Clear">AC</button>
          <button type="button" className="designcolor" id="plusMinus">+/-</button>
          <button type="button" className="designcolor" id="percentage">%</button>
          <button type="button" className="designcolor2" id="divide">&divide;</button>
          <button type="button" className="designcolor" id="seven">7</button>
          <button type="button" className="designcolor" id="eight">8</button>
          <button type="button" className="designcolor" id="nine">9</button>
          <button type="button" className="designcolor2" id="multiply">&times;</button>
          <button type="button" className="designcolor" id="six">6</button>
          <button type="button" className="designcolor" id="five">5</button>
          <button type="button" className="designcolor" id="four">4</button>
          <button type="button" className="designcolor2" id="minus">&minus;</button>
          <button type="button" className="designcolor" id="three">3</button>
          <button type="button" className="designcolor" id="two">2</button>
          <button type="button" className="designcolor" id="one">1</button>
          <button type="button" className="designcolor2" id="one">+</button>
          <button type="button" className="designcolor" id="zero">0</button>
          <button type="button" className="designcolor" id="decimalpoint">&sdot;</button>
          <button type="button" className="designcolor2" id="equals">=</button>
        </div>
      </div>
    );
  }
}
