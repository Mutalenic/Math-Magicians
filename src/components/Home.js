import React from 'react';

const Home = () => (
  <div className="container">
    <h1>
      Welcome To Our Page
    </h1>
    <p>
      Mathematics is an area of knowledge,
      which includes the study of such topics as numbers,
      formulas and related structures,
      shapes and spaces in which they are contained,
      and quantities and their changes.
    </p>
    <p>
      {' '}
      There is no general consensus about its exact scope or epistemological status.
      Wikipedia.
    </p>
  </div>
);

export default Home;
