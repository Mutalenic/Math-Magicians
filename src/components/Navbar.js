import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <div className="navbar">
    <ul>
      <li>
        <Link className="navbar-link math" to="/calculator">Math Magicians</Link>
      </li>
      <li>
        <Link className="navbar-link" to="/">Home</Link>
      </li>
      <li>
        <Link className="navbar-link" to="/calculator">Calculator</Link>
      </li>
      <li>
        <Link className="navbar-link" to="/quote">Quote</Link>
      </li>
    </ul>
  </div>
);

export default Navbar;
