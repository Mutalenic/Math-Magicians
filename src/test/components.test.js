import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Navbar from '../components/Navbar';
import Calculator from '../components/Calculator';
import Home from '../components/Home';
import Quote from '../components/Quotes';
// render tests for components
describe('Test the components', () => {
  test('Calculator render test', () => {
    const renderCal = render(<Calculator />);
    expect(renderCal).toMatchSnapshot();
  });

  test('test render Home', () => {
    const renderHome = render(<Home />);
    expect(renderHome).toMatchSnapshot();
  });

  test('test render Quotes', () => {
    const renderQuote = render(<Quote />);
    expect(renderQuote).toMatchSnapshot();
  });

  // test('Navigation bar render test', () => {
  //     const renderNav = render(<Navbar />);
  //     expect (renderNav).toMatchSnapshot();
  // });
});

// user interaction tests for components
describe('Test user interaction for components', () => {
  test('Navigation bar test', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    const stringElement = screen.getByText('Math Magicians');
    expect(stringElement).toBeInTheDocument();
  });

  test('Calculator test', () => {
    render(
      <BrowserRouter>
        <Calculator />
      </BrowserRouter>,
    );
    const stringElement = screen.getByText('Hai there! Lets do some Math');
    expect(stringElement).toBeInTheDocument();
  });
  test('Home test', () => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );
    const stringElement = screen.getByText('Welcome To Our Page');
    expect(stringElement).toBeInTheDocument();
  });
  test('Quote test', () => {
    render(
      <BrowserRouter>
        <Quote />
      </BrowserRouter>,
    );
    const stringElement = screen.getByText('Math Quote');
    expect(stringElement).toBeInTheDocument();
  });
});
