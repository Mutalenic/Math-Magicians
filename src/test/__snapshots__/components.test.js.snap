// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the components Calculator render test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container2"
        >
          <h2>
            Hai there! Lets do some Math
          </h2>
        </div>
        <div
          class="layout"
        >
          <div
            class="print"
          >
            <h3
              class="printout"
            >
              0
            </h3>
          </div>
          <div
            class="keyboard"
          >
            <button
              class="designcolor"
              id="Clear"
              type="button"
            >
              AC
            </button>
            <button
              class="designcolor"
              id="plusMinus"
              type="button"
            >
              +/-
            </button>
            <button
              class="designcolor"
              id="percentage"
              type="button"
            >
              %
            </button>
            <button
              class="designcolor2"
              id="divide"
              type="button"
            >
              ÷
            </button>
            <button
              class="designcolor"
              id="seven"
              type="button"
            >
              7
            </button>
            <button
              class="designcolor"
              id="eight"
              type="button"
            >
              8
            </button>
            <button
              class="designcolor"
              id="nine"
              type="button"
            >
              9
            </button>
            <button
              class="designcolor2"
              id="multiply"
              type="button"
            >
              x
            </button>
            <button
              class="designcolor"
              id="six"
              type="button"
            >
              6
            </button>
            <button
              class="designcolor"
              id="five"
              type="button"
            >
              5
            </button>
            <button
              class="designcolor"
              id="four"
              type="button"
            >
              4
            </button>
            <button
              class="designcolor2"
              id="minus"
              type="button"
            >
              -
            </button>
            <button
              class="designcolor"
              id="three"
              type="button"
            >
              3
            </button>
            <button
              class="designcolor"
              id="two"
              type="button"
            >
              2
            </button>
            <button
              class="designcolor"
              id="one"
              type="button"
            >
              1
            </button>
            <button
              class="designcolor2"
              id="plus"
              type="button"
            >
              +
            </button>
            <button
              class="designcolor"
              id="zero"
              type="button"
            >
              0
            </button>
            <button
              class="designcolor"
              id="decimalpoint"
              type="button"
            >
              ⋅
            </button>
            <button
              class="designcolor2"
              id="equals"
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container2"
      >
        <h2>
          Hai there! Lets do some Math
        </h2>
      </div>
      <div
        class="layout"
      >
        <div
          class="print"
        >
          <h3
            class="printout"
          >
            0
          </h3>
        </div>
        <div
          class="keyboard"
        >
          <button
            class="designcolor"
            id="Clear"
            type="button"
          >
            AC
          </button>
          <button
            class="designcolor"
            id="plusMinus"
            type="button"
          >
            +/-
          </button>
          <button
            class="designcolor"
            id="percentage"
            type="button"
          >
            %
          </button>
          <button
            class="designcolor2"
            id="divide"
            type="button"
          >
            ÷
          </button>
          <button
            class="designcolor"
            id="seven"
            type="button"
          >
            7
          </button>
          <button
            class="designcolor"
            id="eight"
            type="button"
          >
            8
          </button>
          <button
            class="designcolor"
            id="nine"
            type="button"
          >
            9
          </button>
          <button
            class="designcolor2"
            id="multiply"
            type="button"
          >
            x
          </button>
          <button
            class="designcolor"
            id="six"
            type="button"
          >
            6
          </button>
          <button
            class="designcolor"
            id="five"
            type="button"
          >
            5
          </button>
          <button
            class="designcolor"
            id="four"
            type="button"
          >
            4
          </button>
          <button
            class="designcolor2"
            id="minus"
            type="button"
          >
            -
          </button>
          <button
            class="designcolor"
            id="three"
            type="button"
          >
            3
          </button>
          <button
            class="designcolor"
            id="two"
            type="button"
          >
            2
          </button>
          <button
            class="designcolor"
            id="one"
            type="button"
          >
            1
          </button>
          <button
            class="designcolor2"
            id="plus"
            type="button"
          >
            +
          </button>
          <button
            class="designcolor"
            id="zero"
            type="button"
          >
            0
          </button>
          <button
            class="designcolor"
            id="decimalpoint"
            type="button"
          >
            ⋅
          </button>
          <button
            class="designcolor2"
            id="equals"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test the components test render Home 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h1>
          Welcome To Our Page
        </h1>
        <p>
          Mathematics is an area of knowledge, which includes the study of such topics as numbers, formulas and related structures, shapes and spaces in which they are contained, and quantities and their changes.
        </p>
        <p>
           
          There is no general consensus about its exact scope or epistemological status. Wikipedia.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h1>
        Welcome To Our Page
      </h1>
      <p>
        Mathematics is an area of knowledge, which includes the study of such topics as numbers, formulas and related structures, shapes and spaces in which they are contained, and quantities and their changes.
      </p>
      <p>
         
        There is no general consensus about its exact scope or epistemological status. Wikipedia.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test the components test render Quotes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h1>
          Math Quote
        </h1>
        <p>
          We will always have STEM with us. Some things will drop out of the public eye and go away,
          <br />
          but there will always be science, engineering, and technology.
          <br />
          And there will always, always be mathematics.
          <br />
          — Katherine Johnson, African-American mathematician
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h1>
        Math Quote
      </h1>
      <p>
        We will always have STEM with us. Some things will drop out of the public eye and go away,
        <br />
        but there will always be science, engineering, and technology.
        <br />
        And there will always, always be mathematics.
        <br />
        — Katherine Johnson, African-American mathematician
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
